src/fourcipp/legacy_io/knotvectors.py

```diff
@@ -55,10 +55,14 @@ def read_knotvectors(list_of_lines):
         line = list_of_lines.pop(0)
 
         # Empty line
-        if not line.strip():
-            continue
+        if type(line) == str:
+            if not line.strip():
+                continue
```

Comment:
You gave me a heart attack with `type(something) == type_of_something`  :laughing: 

Nahh just kidding. What about?
```suggestion
        if isinstance(line, str):
            if not line.strip():
                continue
            else:
                line_list = line.split()
        elif isinstance(line, (int, float)):
            line_list = [line]
        elif line is None:
            continue
        else:
            raise TypeError("some message")   
```

Reply:
Thanks for the review - I knew that this was not really nice 🫣